pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-install-hook.sh

```diff
@@ -15,7 +21,7 @@ dotnetInstallHook() {
                 --output "$out/lib/${pname}" \
                 --configuration "@buildType@" \
                 --no-build \
-                --no-self-contained \
```

Comment:
We had this flag here before but not in pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-build-hook.sh . Is it optional and defaults to no being self-contained or do we need to add it to the other file, too. I am confused.

Reply:
By default, `dotnet` build a framework-dependant application, unless specified in project source (`.csproj`) or on command-line as we had before.

But we have the build (`dotnet build`) and install (`dotnet publish`) stages split up. If we build an application as framework-dependant (`--no-self-contained`, or no flag), then we also have to publish it with the same flag, and same for self-contained apps. If we mix up the flags in build and publish, we result in a runtime error.

The fact we didn't have the same flags passed to both dotnet build and dotnet publish was an error, that didn't really surface since we were only building framework-dependant apps, which is the default.

I guess this PR will also need a documentation update, and a note to not ise this for nixpkgs packages (which should probably be always framework-dependant).

Reply:
Not super familiar with the process and what's required, but I will add that I was unable to build #182972 without first ensuring this line was changed to `--self-contained`, or using this PR and enabling the `selfContainedBuild` flag